Remove specific ETag requirements and defer to RFC 9110 (#228)
w3c/lws-protocol: lws10-core/Operations/update-resource.md

Note: This section describes updating a resource's primary content. To update its metadata, see Section 9.3.2.

> LWS servers MUST handle PUT and PATCH requests on resource URIs as modifications to the resource content only, with no default impact on the associated <a>linkset resource</a>. To optionally update both content and metadata in a single atomic operation, clients MAY include Link headers in the PUT/PATCH request to the resource URI and specify the preference 'Prefer: set-linkset' (as defined in RFC 7240). In this case, the server MUST interpret the provided Link headers as a replacement (for PUT) or partial update (for PATCH) to the linkset, in addition to applying the content changes. This behavior is OPTIONAL for servers but, if supported, MUST be invoked explicitly via the Prefer header to prevent unintentional metadata overwrites. Servers that do not support combined updates MUST ignore the preference or respond with 501 Not Implemented.

Reviewer: ```suggestion
LWS servers MUST handle PUT and PATCH requests on resource URIs as modifications to the resource content only, with no default impact on the associated <a>linkset resource</a>. To optionally update both content and metadata in a single atomic operation, clients MAY include `Link` headers in the PUT/PATCH request to the resource URI and specify the preference `'Prefer: set-linkset'` (as defined in [[RFC7240]]). In this case, the server MUST interpret the provided `Link` headers as a replacement (for PUT) or partial update (for PATCH) to the linkset, in addition to applying the content changes. This behavior is OPTIONAL for servers but, if supported, MUST be invoked explicitly via the `Prefer` header to prevent unintentional metadata overwrites. Servers that do not support combined updates MUST ignore the preference or respond with `501 Not Implemented`.
```

> **PUT (replace full resource)** – Send PUT to the resource URI with new full content in the body and matching Content-Type (generally consistent with existing type). PUT is idempotent for existing resources. Clients SHOULD use conditional requests as defined in [[RFC9110]] to avoid overwriting concurrent changes.

Reviewer: ```suggestion
**PUT (replace full resource)** – Send PUT to the resource URI with new full content in the body and matching `Content-Type` (generally consistent with existing type). PUT is idempotent for existing resources. Clients SHOULD use conditional requests as defined in [[RFC9110]] to avoid overwriting concurrent changes.
```

"state": "Connecticut"
}
```
In this example, the client is updating an existing JSON resource at /alice/personalinfo.json. It includes an If-Match header with an entity tag that it received from an earlier GET or HEAD request. The server will compare that to the current entity tag; if they match, it proceeds to replace the content with the JSON provided. If they don’t match, the server rejects the update (because the resource was changed by someone else in the meantime).

> Successful response: If the update succeeds, the server can respond with 200 OK and possibly include the updated representation or some confirmation (like the new content or a part of it). Alternatively, the server may respond with 204 No Content to indicate success with no body (especially common if no further info needs to be conveyed). For example:

Reviewer: ```suggestion
In this example, the client is updating an existing JSON resource at `/alice/personalinfo.json`. It includes an `If-Match` header with an entity tag (`ETag`) that it received from an earlier GET or HEAD request. The server will compare that to the current entity tag (`ETag`); if they match, it proceeds to replace the content with the JSON provided. If they don’t match, the server rejects the update (because the resource was changed by someone else in the meantime).
Successful response: If the update succeeds, the server can respond with `200 OK` and possibly include the updated representation or some confirmation (like the new content or a part of it). Alternatively, the server may respond with `204 No Content` to indicate success with no body (especially common if no further info needs to be conveyed). For example:
```

```
HTTP/1.1 204 No Content
```

> * **Error responses:** If the `If-Match` precondition did not match (concurrent modification), the server responds with `412 Precondition Failed`. If the resource did not exist, a PUT meant as an update will result in `404 Not Found`. If the client is not authorized, `403 Forbidden` (or 401 Unauthorized if no valid credentials were provided). If the request payload is not valid, `400 Bad Request`.

Reviewer: ```suggestion
* **Error responses:** If the `If-Match` precondition did not match (concurrent modification), the server responds with `412 Precondition Failed`. If the resource did not exist, a PUT meant as an update will result in `404 Not Found`. If the client is not authorized, `403 Forbidden` (or `401 Unauthorized` if no valid credentials were provided). If the request payload is not valid, `400 Bad Request`.
```

Partial Update (PATCH): A PATCH request to the <a>linkset resource</a> URI adds, removes, or modifies specific links.

**Concurrency Control for Metadata**

> Because a resource's metadata can be modified by multiple actors, preventing concurrent overwrites is important. Servers and clients SHOULD use conditional requests as defined in [[RFC9110]] for PUT and PATCH operations on a <a>linkset resource</a>. When a server receives a conditional request for which the precondition fails, it MUST reject the request with a 412 Precondition Failed status code.

Reviewer: ```suggestion
Because a resource's metadata can be modified by multiple actors, preventing concurrent overwrites is important. Servers and clients SHOULD use conditional requests as defined in [[RFC9110]] for PUT and PATCH operations on a <a>linkset resource</a>. When a server receives a conditional request for which the precondition fails, it MUST reject the request with a `412 Precondition Failed` status code.
```
